timestamp-impl/src/main/java/com/palantir/timestamp/TimestampAllocationFailures.java

```diff
@@ -49,17 +50,17 @@ public RuntimeException responseTo(Throwable newFailure) {
         }
 
         if (newFailure instanceof MultipleRunningTimestampServiceError) {
+            encounteredMultipleRunningTimestamps = true;
             return wrapMultipleRunningTimestampServiceError(newFailure);
         }
 
         return new RuntimeException("Could not allocate more timestamps", newFailure);
     }
 
-    public void verifyWeShouldTryToAllocateMoreTimestamps() {
+    public void verifyWeShouldIssueMoreTimestamps() {
         // perform only single volatile load on hot path
-        Throwable failure = this.previousAllocationFailure;
-        if (failure instanceof MultipleRunningTimestampServiceError) {
-            throw wrapMultipleRunningTimestampServiceError(failure);
+        if (encounteredMultipleRunningTimestamps) {
+            throw wrapMultipleRunningTimestampServiceError(this.previousAllocationFailure);
         }
     }
```

Comment:
I think it's correct. I have a further optimisation I'd like to test out at some point, but let's get this in first.